src/dfmt/formatter.d

```diff
@@ -1868,7 +1873,8 @@ private:
             else
                 formatStep();
         }
-        while (index < tokens.length && parenDepth > 0);
+        // TODO: obviously getting stuck here?
+        while (hasCurrent && parenDepth > 0);
```

Comment:
not quite sure how this isn't getting stuck before? Both are only getters, not modifying the state?

Reply:
Are you saying that you think that this is an infinite loop? `formatStep` and `writeToken` calls cause later calls to `hasCurrent` to return a different value.

Reply:
ohh nvm I didn't notice this was a do-while loop, I thought it was an empty while loop